### ISSUE TITLE
Shuffle vertical JointDataset samples

### ISSUE THREAD

**Issue by nkemnitz** [ZettaAI/zetta_utils#324]
Current behavior: When training on a vertically JointDataset, the datasets are processed in order of insertion and thus samples are drawn from a single dataset (with likely similar statistics) before moving on to the next dataset.

Feature request: A way to allow sampling across all datasets within a JointDataset. Will create a PR with my current solution, but maybe there is already another way to do it that I haven't figured out?

**Comment by supersergiy** [comment]
I'm assuming setting [[shuffle](https://github.com/ZettaAI/zetta_utils/blob/main/specs/sergiy/training/encoding_coarsener/gen_x1/0_32nm_128nm.cue#L138)] parameter in the torch.DataLoader doesn't work for your usecase because the joint dataset is not used directly for training, bur rather is a building block that's joint vertically/horizontally again later?
